diff --git a/src/corelib/global/qlibraryinfo.cpp b/src/corelib/global/qlibraryinfo.cpp
index 301c3ab8..bfb28380 100644
--- a/src/corelib/global/qlibraryinfo.cpp
+++ b/src/corelib/global/qlibraryinfo.cpp
@@ -624,10 +624,11 @@ static QString getPrefix(
 #  if QT_CONFIGURE_CROSSBUILD
     if (group == QLibraryInfo::DevicePaths)
         return QString::fromLocal8Bit(QT_CONFIGURE_PREFIX_PATH);
+#  elif 0 //QT_CONFIG(relocatable)
+    return getExtPrefixFromHostBinDir();
 #  else
-    Q_UNUSED(group);
+    return QString::fromLocal8Bit(QT_CONFIGURE_PREFIX_PATH);
 #  endif
-    return getExtPrefixFromHostBinDir();
 #elif QT_CONFIG(relocatable)
     return getRelocatablePrefix();
 #else
