diff --git a/src/gui/kernel/qguiapplication.cpp b/src/gui/kernel/qguiapplication.cpp
index 08a00d82..f555fb4f 100644
--- a/src/gui/kernel/qguiapplication.cpp
+++ b/src/gui/kernel/qguiapplication.cpp
@@ -1411,13 +1411,7 @@ void QGuiApplicationPrivate::createPlatformIntegration()
     bool isGnome = false;
     const QByteArray waylandPlatformName = defaultIsWayland ? platformName : "wayland";
     if (hasWaylandDisplay || isWaylandSessionType) {
-        const QByteArray currentDesktop = qgetenv("XDG_CURRENT_DESKTOP").toLower();
-        const QByteArray sessionDesktop = qgetenv("XDG_SESSION_DESKTOP").toLower();
-        isGnome = currentDesktop.contains("gnome") || sessionDesktop.contains("gnome");
-        if (isGnome)
-            preferredPlatformOrder.append(waylandPlatformName);
-        else
-            preferredPlatformOrder.prepend(waylandPlatformName);
+        preferredPlatformOrder.prepend(waylandPlatformName);
 
         if (defaultIsWayland)
             platformName.clear();
